SocialCareCaseViewerApi.Tests/V1/UseCase/SubmissionUseCaseTests.cs

```diff
@@ -0,0 +1,90 @@
+        private const string CollectionName = "resident-case-submissions";
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockDatabaseGateway = new Mock<IDatabaseGateway>();
+            _mockMongoGateway = new Mock<IMongoGateway>();
+            _submissionsUseCase = new SubmissionsUseCase(_mockDatabaseGateway.Object, _mockMongoGateway.Object);
+        }
+
+        [Test]
+        public void ExecutePostSuccessfully()
+        {
+            var request = TestHelpers.CreateCaseSubmissionRequest();
+            var worker = TestHelpers.CreateWorker();
+            var resident = TestHelpers.CreatePerson();
+
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy)).Returns(worker);
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(request.ResidentId)).Returns(resident);
+            _mockMongoGateway.Setup(x => x.InsertRecord(It.IsAny<string>(), It.IsAny<CaseSubmission>()));
+
+            var (caseSubmissionResponse, caseSubmission) = _submissionsUseCase.ExecutePost(request);
+            var expectedResponse = TestHelpers.CreateCaseSubmission(SubmissionState.InProgress,
+                caseSubmission.CreatedAt, worker, resident, caseSubmission.SubmissionId, request.FormId);
+
+            caseSubmissionResponse.Should().BeEquivalentTo(expectedResponse.ToDomain().ToResponse());
+            _mockDatabaseGateway.Verify(x => x.GetWorkerByEmail(request.CreatedBy), Times.Once);
+            _mockDatabaseGateway.Verify(x => x.GetPersonByMosaicId(request.ResidentId), Times.Once);
+            _mockMongoGateway.Verify(x => x.InsertRecord(CollectionName, caseSubmission), Times.Once);
+        }
+
+        [Test]
+        public void ExecutePostThrowsAnErrorWhenNoWorkerFoundForRequest()
+        {
+            var request = TestHelpers.CreateCaseSubmissionRequest();
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy));
+
+            Action act = () => _submissionsUseCase.ExecutePost(request);
+
+            act.Should().Throw<WorkerNotFoundException>()
+                .WithMessage($"Worker with email {request.CreatedBy} not found");
+        }
+
+        [Test]
+        public void ExecutePostThrowsAnErrorWhenNoResidentFoundForRequest()
+        {
+            var request = TestHelpers.CreateCaseSubmissionRequest();
+            var worker = TestHelpers.CreateWorker();
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy)).Returns(worker);
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(request.ResidentId));
+
+            Action act = () => _submissionsUseCase.ExecutePost(request);
+
+            act.Should().Throw<PersonNotFoundException>()
+                .WithMessage($"Person with id {request.ResidentId} not found");
+        }
+
+        [Test]
+        public void ExecuteGetByIdSucessfully()
+        {
+            var submissionResponse = TestHelpers.CreateCaseSubmission();
```

Comment:
I think it would be nice if we split test into 3 blocks using the arrange, act, assert pattern.
It makes our tests very readable.

Arrange is create any variables you need, mock any functions that require mock.

Act is call the actual tested function and store the response in a variable.

Assert is our tests so checking mocked function or the value is what we expect.